src/main/features/claimant-response/routes/payment-plan.ts

```diff
@@ -104,24 +116,33 @@ class PaymentPlanPage extends AbstractPaymentPlanPage<DraftClaimantResponse> {
 
     if (decisionType === DecisionType.COURT) {
       const paymentPlanFromDefendantFinancialStatement: PaymentPlan = PaymentPlanHelper.createPaymentPlanFromDefendantFinancialStatement(claim)
-      const lastPaymentDate: Moment = paymentPlanFromDefendantFinancialStatement.calculateLastPaymentDate()
-
-      courtCalculatedPaymentIntention.paymentDate = lastPaymentDate
-      courtCalculatedPaymentIntention.paymentOption = draft.document.alternatePaymentMethod.toDomainInstance().paymentOption
-      courtCalculatedPaymentIntention.repaymentPlan = {
-        firstPaymentDate: paymentPlanFromDefendantFinancialStatement.startDate,
-        instalmentAmount: paymentPlanFromDefendantFinancialStatement.instalmentAmount,
-        paymentSchedule: Frequency.toPaymentSchedule(paymentPlanFromDefendantFinancialStatement.frequency),
-        completionDate: paymentPlanFromDefendantFinancialStatement.calculateLastPaymentDate(),
-        lengthOfPayment: paymentPlanFromDefendantFinancialStatement.calculatePaymentLength()
+      const defendantFrequency: Frequency = PaymentSchedule.toFrequency(claimResponse.paymentIntention.repaymentPlan.paymentSchedule)
+      const paymentPlanConvertedToDefendantFrequency: PaymentPlan = paymentPlanFromDefendantFinancialStatement.convertTo(defendantFrequency)
+
+      if (claimResponse.paymentIntention.paymentOption === PaymentOption.INSTALMENTS) {
+        courtCalculatedPaymentIntention.paymentOption = PaymentOption.INSTALMENTS
```

Comment:
construction of `courtCalculatedPaymentIntention` also duplicated in few place.

Reply:
yes, i know. Once paymentPlan has completion date and length of payment we can just do 

courtCalculatedPaymentIntention.repaymentPlan = {whatever repayment plan }

instead of calculating everything everytime with different values. 

The code as it stands can't be refactored out. Even if possible, its not worth doing now and again refactor when completion date and length of payment changes are done.